### ISSUE TITLE
date_add field is wrongly modified when updating a product

### ISSUE THREAD

**Issue by Jeremy-VERNET** [PrestaShop/PrestaShop#21707]
#### Describe the bug

The site had errors in displaying products marked as "New".

I wanted to see what was happening in the database and I noticed that each time I modified a product, it correctly modified the `date_upd` field but completely changed `date_add` of the ps_products table.

#### Expected behavior

Dates should change normally.

**Screenshots**

Here is a little gif showing the bug: https://imgur.com/a/jl9KNor

I show you the correct date_add and when I update the product, it changes it to a 2005 date.

And here is the correlation between the update time (which is correct) and the date_add. I can't find any pattern...

https://i.imgur.com/ZpsjU9l.png

#### Additional information

* PrestaShop version: 1.7.6.7
* PHP version: 7.1

**Comment by hibatallahAouadni** [comment]
Hello @Jeremy-VERNET 

I didn't manage to reproduce your issue with the provided information.
I attached a screen record:
https://drive.google.com/file/d/1APFHi3h4U625RXmOrVQCRoriyrvHYQgt/view)

Thanks to check and feedback.

**Comment by Jeremy-VERNET** [comment]
Thank you for trying @hibatallahAouadni 

What is strange is that this kind of problem has nothing to do with a theme or a module since it is a "native" function of Prestashop.

Regarding my version of Prestashop, I did a clean installation locally, I imported data from a 1.7 Beta version, everything worked fine so I put everything in production. The store is currently in production and everything works until this error...

**Comment by SimonGrn** [comment]
It's a native function of PrestaShop, and this works well with a fresh installation... Are you sure your shop doesn't have any module that could interfere with that ?
We need to be able to reproduce the issue to validate it.

**Comment by hibatallahAouadni** [comment]
>I imported data from a 1.7 Beta version

How did you import the data from 1.7 Beta version? and could you please specify what Beta version you're talking about? 1.7.7.0 Beta or 1.7.7.0 Beta 2?

>this kind of problem has nothing to do with a theme or a module since it is a "native" function of Prestashop.

Sometimes, an installed module could override a native function. So please check if you have an override.

Thanks!

**Comment by Jeremy-VERNET** [comment]
@SimonGrn : Of course, I understand. I only have 3 modules (except those already included by Prestashop) : PayPal (202 ecommerce), Stripe (202 ecommerce), Boxtal Connect (Boxtal). These are just well known modules and they are all updated.

@hibatallahAouadni : The old Prestashop was 1.7.1.1 and I used [this page](https://devdocs.prestashop.com/1.7/basics/keeping-up-to-date/migration/) to have an idea on how to migrate. I exported each table in CSV then imported each table to the new one.

I would like to point out that during the import phase, I only imported the content and not the complete table. So the structure of the table is that of the new version.

I also checked all the folders present in the /override/ folder and I don't have a single override.

**Comment by SimonGrn** [comment]
I'll ask the dev team to take a look into it. Stay tuned.

Poke @PrestaShop/prestashop-maintainers what do you think could be the reason ?

**Comment by matks** [comment]
Hi @Jeremy-VERNET to the best of my knowledge, the issue does not come the core.

I suggest you check this part of the code: line 722 class ObjectModel https://github.com/PrestaShop/PrestaShop/blob/develop/classes/ObjectModel.php#L722
```
        // Automatically fill dates
        if (property_exists($this, 'date_add') && $this->date_add == null) {
            $this->date_add = date('Y-m-d H:i:s');
            if (isset($this->update_fields) && is_array($this->update_fields) && count($this->update_fields)) {
                $this->update_fields['date_add'] = true;
            }
        }
```

As you can see it's the part responsible for filling `date_add` property. So by debugging this part (either insert a var_dump or use a step-by-step debugger with xdebug) you should see what's wrong.

**Comment by SimonGrn** [comment]
Thank you @matks !
I will close this issue since it's not related to the Core.
You can still answer and give details if you find the reason !

Thank you for your understanding.

**Comment by Jeremy-VERNET** [comment]
For those that can help, the problem is now resolved.

**1)** The "New" label did not disappear because I modified the date of the products in _ps_product_ but the one that is taken into account for the label is _ps_product_shop_.

**2)** Then, for the problem where with each modification, it modifies the dates oddly, it is surely due to the differences of dates which were present (some in 2031, others in 2004...). By correcting all dates to a current date, this problem is also resolved.
